### PR TITLE
x265: fix cross compilation to non-aarch64

### PR DESCRIPTION
**Pull request by alyssais**
###### Description of changes

Previously, the multibitdepth cmake invocations didn't use cmakeFlags.
This meant that they didn't get the flags added by stdenv for cross
compilation, and so they didn't cross compile.  To fix this, add
cmakeCommonFlags into cmakeFlags, and then add all the non-common
flags to cmakeFlagsArray after the multibitdepth stuff has been built.

As a consequence of this change, we also no longer need to disable asm
for RISC-V.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/x265/default.nix [hunk 1]
```diff
@@ -96,32 +76,54 @@ stdenv.mkDerivation rec {
...
+  '' + ''
+    cmakeFlagsArray+=(
+      -DGIT_ARCHETYPE=1 # https://bugs.gentoo.org/814116
+      ${mkFlag stdenv.hostPlatform.isStatic "ENABLE_SHARED"}
+      -DHIGH_BIT_DEPTH=OFF
+      -DENABLE_HDR10_PLUS=ON
+      ${mkFlag (isCross && stdenv.hostPlatform.isAarch) "CROSS_COMPILE_ARM"}
+      ${mkFlag cliSupport "ENABLE_CLI"}
+      ${mkFlag unittestsSupport "ENABLE_TESTS"}
+    )
```
Comment (NickCao) [inline pkgs/development/libraries/x265/default.nix:124]: What's the point of setting `cmakeFlagsArray` here, weren't all the configure already done in `configurePhase`?

Comment (alyssais) [reply pkgs/development/libraries/x265/default.nix:124]: Yeah that's a mistake.  Thank you for noticing.

**Comment by ashkitten** [comment]
i used this as a base for #253225, and added some fixes in 8fc2a34ab8b866729fadf226864e885d872fd380 to get it actually working correctly

**Comment by alyssais** [comment]
Sorry for taking so long, but I think this is ready now?  There's a good chance I've missed something, because it's been so long, so @ashkitten @Artturin I'd appreciate it if you could take a look.

**Comment by Artturin** [comment]
Diffed the output and no diff

**Pull request merged**